trino/auth.py

```diff
@@ -142,101 +143,77 @@ class _OAuth2TokenBearer(AuthBase):
     """
     MAX_OAUTH_ATTEMPTS = 5
 
-    class _AuthStep(Enum):
-        GET_REDIRECT_SERVER = 0
-        GET_TOKEN = 1
-
     def __init__(self, http_session, redirect_auth_url_handler=handle_redirect_auth_url):
         self._redirect_auth_url = redirect_auth_url_handler
-        self.http_session = http_session
         self._thread_local = threading.local()
+        self._pattern = re.compile(r'Bearer x_redirect_server="(.+)", x_token_server="(.+)"')
```

Comment:
What if order is different (or lowercase/uppercase)? I think the older approach of split was more robust.

Reply:
The previous approach didn't handle not matching headers well. I've reverted to the old split method but added an additional check.